### DISCUSSION TITLE
barrier.config -- different synaxes/versions ?

### DISCUSSION THREAD

**Discussion by DokBiz** [debauchee/barrier#1594] [Q&A]
my barrier server (auto)starts up with the promt: 
"Failed to get local IP address. Please manually type in server address on your clients"
i need to press "ok" and anything works fine then.. 
now i want to avoid pressing "ok" everytime i start my server and start to investigate. thereby i realized the server says: "IP-addresses: unknown", the client shows up his own IP here (and a couple of mysterious others, but this is not part of the actual problem.. i saw few days ago discussions for this somewhere)

anyway, i took a look into the Barrier.conf and noticed the Barrier.conf that comes with the standard installation differs significantly with the .example files (/usr/share/doc/barrier-headless/barrier.conf.example) or in any wiki.

the standard installation one:

```
[General]
autoConfig=true
autoConfigPrompted=true
autoHide=true
autoStart=true
...

[internalConfig]
clipboardSharing=true
enableDragAndDrop=true
hasHeartbeat=false
hasSwitchDelay=false
hasSwitchDoubleTap=false
heartbeat=5000
hotkeys\size=0
ignoreAutoConfigClient=false
numColumns=5
numRows=3
relativeMouseMoves=false
screenSaverSync=true
screens\1\name=
screens\10\name=
screens\11\name=
screens\12\name=
screens\13\name=
...
```

compared to:

```
# sample barrier configuration file
#
# comments begin with the # character and continue to the end of
# line.  comments may appear anywhere the syntax permits.

section: screens
	# three hosts named:  moe, larry, and curly
	moe:
	larry:
	curly:
end

section: links
	# larry is to the right of moe and curly is above moe
	moe:
		right = larry
		up    = curly

	# moe is to the left of larry and curly is above larry.
	# note that curly is above both moe and larry and moe
	# and larry have a symmetric connection (they're in
	# opposite directions of each other).
	larry:
		left  = moe
		up    = curly

	# larry is below curly.  if you move up from moe and then
	# down, you'll end up on larry.
	curly:
		down  = larry
end

section: aliases
	# curly is also known as shemp
	curly:
		shemp
end

```

so i'm a bit confused now !? what is the right one, or am i missing something?
do i need to add the example part underneath the existing part, or do i need to exchange completely, or do i need different files?

**Comment by tonymynd** [comment]
no answer uh?

**Reply by poleguy** [reply]
I just tried to ask a similar question in the IRC chat, but haven't heard an answer yet. I think the answer to your question is wrapped up in the barrier command that seems to be a wrapper. Maybe this info will help inform someone who can answer this question...

# https://web.libera.chat/

Good day... I wanted to try to debug the issue of the screensaver not activating on my client. However I can't find any documentation of the barrier command (not barriers or barrierc) so I can't figure out how to start barrier with the --debug DEBUG flag to see the log messages. How do I get --debug DEBUG set on ubuntu?

My problem specifically is: If I run barrier --debug DEBUG, it ignores the flag and runs barriers indirectly with these settings:
("-f", "--no-tray", "--debug", "INFO", "--name", "elgar", "--disable-client-cert-checking", "-c", "/tmp/Barrier.xeUdSy", "--address", ":24800")

This makes me think the barrier command is just a wrapper. However I can't find documentation for it or source code to understand its magic. I see source for barriers and barrierc, but not barrier itself.
If I then try running barriers myself, I get this, because I don't know where the barrier.conf file exists... the one in /tmp/Barrier.xxxxxx seemingly created magically by the barrier wrapper:
barriers --debug DEBUG --name elgar --address :24800 -f --no-tray --disable-client-cert-checking
[2023-09-13T07:29:38] DEBUG: opening configuration "/home/poleguy/.local/share/barrier/.barrier.conf"
[2023-09-13T07:29:38] DEBUG: cannot open configuration "/home/poleguy/.local/share/barrier/.barrier.conf"
[2023-09-13T07:29:38] DEBUG: opening configuration "/etc/barrier.conf"
[2023-09-13T07:29:38] DEBUG: cannot open configuration "/etc/barrier.conf"
barriers: no configuration available

I mean, I can hack it by just specifying a stale /tmp/Barrier.xxxxxx, but it bothers me that I don't understand where that file is coming from and that there is seemingly no documentation of how it's supposed to work. It seems there were are proper man pages for barrierc and barriers, but not for barrier. I guess barrier is a wrapper somebody added after the fact and didn't get around to documenting?

The motivating problem is that the screensaver seems to not be detected. Looking in the source I'm getting the impression that xscreensaver is supported, but maybe not xfce4-screensaver? Still, I'd like to at least figure out how to get barrier to not break xfce4-screensaver on the client, even if it isn't syrchorized.
